include/revng/Model/model-schema.yml

```diff
@@ -245,469 +349,915 @@ definitions:
-        doc: The type of this `union` alternative.
         upcastable: true
-    key:
-      - Index
-  - name: StructField
+
+  #
+  # Type wrapping system (pointers / arrays / primitives)
+  #
+
+  - name: Type
+    type: struct
+
     doc: |-
-      A field of a `StructDefinition`.
+      A type such as an array, a pointer, a primitive or a defined type.
 
-      It is composed by the offset of the field and its type.
+    abstract: true
+
+    fields:
+      - name: Kind
+        doc: |-
+          Specifies which of the children classes this object is upcastable to.
+        type: TypeKind
+
+      - name: IsConst
+        doc: Specifies whether the type this object denotes is `const` or not.
+        type: bool
+        optional: true
+
+  - name: PrimitiveType
+    inherits: Type
     type: struct
+
+    doc: |-
+      A Primitive type defined by its kind (signed, float, etc) and size.
+
     fields:
-      - name: Offset
+      - name: PrimitiveKind
+        doc: |-
+          Specifies whether this primitive type is `Signed`, `Float`,
+          `PointerOrNumber`, and so on.
+        type: PrimitiveKind
+
+      - name: Size
+        doc: |-
+          Specifies the size of this primitive type.
+
+          As of now, supported sizes include
+          ```
+          { 1, 2, 4, 8, 16 }
+          ```
+
+          Note that *only* floating point primitives (and `Generic` primitives,
+          whose kind is not known, so they are assumed to be floating point
+          compatible) can also have additional sizes:
+          ```
+          { 10, 12 }
+          ```
+
+          Note that 1-byte floating point primitives are not allowed.
+
+          Note that `Void` *must* have size of 0..
         type: uint64_t
-      - name: Name
-        type: string
-        optional: true
-      - name: Comment
-        type: string
         optional: true
-      - name: Type
+
+  - name: PrimitiveKind
+    type: enum
+
+    doc: |-
+      The kind of a primitive type.
+
+    members:
+      - name: Void
+        doc: The `void` type.
+
+      - name: Generic
+        doc: |-
+          The most generic primitive kind: it can be any of the other primitive
+          kinds, except for `Void`.
+
+      - name: PointerOrNumber
+        doc: |-
+          A kind representing either a `Number` kind or a pointer.
+          This can also be seen as not-a-`Float`.
+
+      - name: Number
+        doc: |-
+          A two's complement integer number, either `Signed` or `Unsigned`.
+
+      - name: Unsigned
+        doc: |-
+          An `unsigned` two's complement integer number.
+
+      - name: Signed
+        doc: |-
+          An `signed` two's complement integer number.
+
+      - name: Float
+        doc: |-
+          A IEEE 754 floating point number.
+
+  - name: PointerType
+    inherits: Type
+    type: struct
+
+    doc: A pointer type.
+
+    fields:
+      - name: PointerSize
+        doc: As of now, only 4 and 8 byte pointers are supported.
```

Comment:
Add a preliminary obvious comment such as "The size of the pointer".

Reply:
Done!